contracts/SmartLoan.sol

```diff
@@ -0,0 +1,459 @@
+    governor = _governor;
+    MAX_LTV = 5000;
+    MIN_SELLOUT_LTV = 4000;
+  }
+
+
+  /**
+   * Funds a loan with the value attached to the transaction
+  **/
+  function fund() external payable {
+
+    emit Funded(msg.sender, msg.value, block.timestamp);
+  }
+
+
+  function setMaxLTV(uint256 _newMaxLtv) external {
+    require(msg.sender == governor, "Only the governor account can change the maximal LTV");
+    MAX_LTV = _newMaxLtv;
+  }
+
+
+  function setMinSelloutLTV(uint256 _newMinSelloutLtv) external {
+    require(msg.sender == governor, "Only the governor account can change the minimal sellout ltv");
+    MIN_SELLOUT_LTV = _newMinSelloutLtv;
+  }
+
+
+  /**
+   * This function allows selling assets without checking if the loan will remain solvent after this operation.
+   * It is used as part of the sellout() function which sells part/all of assets in order to bring the loan back to solvency.
+   * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventAssetSale(bytes32 asset, uint256 _amount, uint256 _minAvaxOut) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    token.transfer(address(exchange), _amount);
+
+    (bool success,) = address(exchange).call{value : 0}(
+      abi.encodeWithSignature("sellAsset(bytes32,uint256,uint256)", asset, _amount, _minAvaxOut)
+    );
+    if (!success) {
+      exchange.transferBack(asset);
+    }
+  }
+
+
+  /**
+  * This function attempts to sell just enough asset to receive targetAvaxAmount.
+  * If there is not enough asset's balance to cover the whole targetAvaxAmount then the whole asset's balance
+  * is being sold.
+  * It is possible that multiple different assets will have to be sold and for that reason we do not use the remainsSolvent modifier.
+  **/
+  function nonSolventPartialOrFullAssetSale(bytes32 asset, uint256 targetAvaxAmount) private {
+    IERC20Metadata token = getERC20TokenInstance(asset);
+    uint256 balance = token.balanceOf(address(this));
+    if (balance > 0) {
+      uint256 minSaleAmount = exchange.getMinimumERC20TokenAmountForExactAVAX(targetAvaxAmount, supportedAssets.getAssetAddress(asset));
+      if (balance < minSaleAmount) {
+        uint256 saleAvaxValue = exchange.getEstimatedAVAXFromERC20Token(balance, supportedAssets.getAssetAddress(asset));
+        nonSolventAssetSale(asset, balance, saleAvaxValue);
+      } else {
+        nonSolventAssetSale(asset, minSaleAmount, targetAvaxAmount);
+      }
+    }
+  }
+
+
+  /**
+  * This function attempts to repay the _repayAmount back to the pool as well as pay a bonus to liquidator.
+  * If there is not enough AVAX balance to repay the _repayAmount then the available AVAX balance will be repaid and no
+  * liquidation bonus will be paid to the liquidator.
+  **/
+  function attemptRepay(uint256 _repayAmount) internal {
+    if (address(this).balance <= _repayAmount) {
```

Comment:
We've got 5 LOC to do something trivial how about
repay(Math.min(address(this).balance, _repayAmount)); ?

https://docs.openzeppelin.com/contracts/3.x/api/math#Math-min-uint256-uint256-

Reply:
Good idea